pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -410,9 +408,8 @@ func keepCurrentContainerResources(declared, current *appsv1.Deployment) bool {
 }
 
 // deployment returns the deployment from the server
-func (r *reconcilerBase) deployment(ctx context.Context, dRef client.ObjectKey) (*appsv1.Deployment, error) {
-	depObj := &appsv1.Deployment{}
-	depUnObj, err := r.dynamicClient.Resource(kinds.DeploymentResource()).Namespace(dRef.Namespace).Get(ctx, dRef.Name, metav1.GetOptions{})
+func (r *reconcilerBase) deployment(ctx context.Context, dRef client.ObjectKey) (*unstructured.Unstructured, error) {
+	unstructuredDeployment, err := r.dynamicClient.Resource(kinds.DeploymentResource()).Namespace(dRef.Namespace).Get(ctx, dRef.Name, metav1.GetOptions{})
```

Comment:
nit: This could remain depObj or be deployObj to be consistent with the other method. But it's not super important.